### PR TITLE
mod-admin-full: Allow odhcpd to be configured via LuCI

### PR DESCRIPTION
**Pull request by cshoredaniel**
Add odhcp section configuration and per-interface odhcpd for v4.

Signed-off-by: Daniel Dickinson <lede@cshore.thecshore.com>

### CODE REVIEW

**Comment by danrl** [comment]
I have something similar in the works which also is fully IPv6 capable. May I suggest that I create PR for my changes and we then pick the best of both?

**Comment by cshoredaniel** [comment]
Makes sense to me!

**Comment by danrl** [comment]
Similar PR here: https://github.com/openwrt/luci/pull/990

File: modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/dhcp.lua [hunk 1]
```diff
@@ -255,6 +255,22 @@ o = s:taboption("general", DynamicList, "notinterface",
 o.optional = true
 o:depends("nonwildcard", true)
 
+s = m:section(NamedSection, "odhcpd", "odhcpd", translate("odhcpd Server Settings"))
```
Comment (danrl) [inline modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/dhcp.lua]: Does this test for odhcpd being available?
Are there official builds without odhcpd? If so, we might want to check which of one (or both) of the dhcp-servers are available.

E.g. like this: https://github.com/openwrt/luci/pull/990/files#diff-aed351d2580a213ca424f7a09aaa8a43R15

**Comment by danrl** [comment]
I like the changes in `dhcp.lua`. I think `ifaces.lua` is missing some features of odhcpd.

**Comment by cshoredaniel** [comment]
Marking this and #990 as 'needs update' to reflect desire to merge the two.

**Comment by cshoredaniel** [comment]
Closing as have taken a break from LEDE/OpenWrt development for various reasons, so will reopen when this is back on the radar.

**Pull request closed without merging**